pkg/backup/backup_new_test.go

```diff
@@ -1178,6 +1254,382 @@ func TestBackupActionAdditionalItems(t *testing.T) {
 	}
 }
 
+// volumeSnapshotterGetter is a simple implementation of the VolumeSnapshotterGetter
+// interface that returns velero.VolumeSnapshotters from a map if they exist.
+type volumeSnapshotterGetter map[string]velero.VolumeSnapshotter
+
+func (vsg volumeSnapshotterGetter) GetVolumeSnapshotter(name string) (velero.VolumeSnapshotter, error) {
+	snapshotter, ok := vsg[name]
+	if !ok {
+		return nil, errors.New("volume snapshotter not found")
+	}
+
+	return snapshotter, nil
+}
+
+func int64Ptr(val int) *int64 {
+	i := int64(val)
+	return &i
+}
+
+type volumeIdentifier struct {
+	volumeID string
+	volumeAZ string
+}
+
+type volumeInfo struct {
+	volumeType  string
+	iops        *int64
+	snapshotErr bool
+}
+
+// fakeVolumeSnapshotter is a test fake for the velero.VolumeSnapshotter interface.
+type fakeVolumeSnapshotter struct {
+	// PVVolumeNames is a map from PV name to volume ID, used as the basis
+	// for the GetVolumeID method.
+	PVVolumeNames map[string]string
+
+	// Volumes is a map from volume identifier (volume ID + AZ) to a struct
+	// of volume info, used for the GetVolumeInfo and CreateSnapshot methods.
+	Volumes map[volumeIdentifier]*volumeInfo
+}
+
+// WithVolume is a test helper for registering persistent volumes that the
+// fakeVolumeSnapshotter should handle.
+func (vs *fakeVolumeSnapshotter) WithVolume(pvName, id, az, volumeType string, iops int, snapshotErr bool) *fakeVolumeSnapshotter {
+	if vs.PVVolumeNames == nil {
+		vs.PVVolumeNames = make(map[string]string)
+	}
+	vs.PVVolumeNames[pvName] = id
+
+	if vs.Volumes == nil {
+		vs.Volumes = make(map[volumeIdentifier]*volumeInfo)
+	}
+
+	identifier := volumeIdentifier{
+		volumeID: id,
+		volumeAZ: az,
+	}
+
+	vs.Volumes[identifier] = &volumeInfo{
+		volumeType:  volumeType,
+		iops:        int64Ptr(iops),
+		snapshotErr: snapshotErr,
+	}
+
+	return vs
+}
+
+// Init is a no-op.
+func (vs *fakeVolumeSnapshotter) Init(config map[string]string) error {
+	return nil
```

Comment:
Whenever `vs` is not being used inside the function, it can be omitted. Makes it easier to know it's not used. The functions that call panic down below can also have this omitted. Not a big deal but I think we should aim to do this for the code base.

Reply:
done